lib/datadog/statsd/forwarder.rb

```diff
@@ -22,24 +20,20 @@ def initialize(
 
         logger: nil
       )
-        @transport_type = socket_path.nil? ? :udp : :uds
+        @transport_type = connection_cfg.transport_type
```

Comment:
Should we test `connection_cfg.nil?` here? Especially since it's the constructor default value.

Reply:
I think this is where required keyword arguments, mentioned above, would be helpful.  It's basically required without the language enforcing that.

Reply:
There's a "trick" to simulate the same behavior as required keyword args:

```ruby
def foo(bar: raise(ArgumentError, 'missing keyword :bar'))
```

But it's probably not a bad idea to consider dropping Ruby 2.0 support in the near future (dd-trace-rb already did it! We're living the good life on 2.1+).

Reply:
Dropping 2.0 is planned:
https://github.com/DataDog/dogstatsd-ruby/blob/ed04ca4ac4543c409a6fb2a5014c36fcc125371d/lib/datadog/statsd.rb#L107-L109

Reply:
🥳 🎉 🎈